platforms/software/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/report/generic/MetadataRendererRegistry.java

```diff
@@ -123,16 +148,36 @@ public SimpleHtmlWriter render(Object metadata, SimpleHtmlWriter htmlWriter) thr
         }
     }
 
-    public static final class URIMetadataRenderer implements MetadataRenderer<URI> {
+    public static final class ClickableLinkRenderer implements MetadataRenderer {
+        private final Set<Class<?>> metadataTypes = ImmutableSet.of(URI.class, File.class, RegularFile.class);
+
         @Override
-        public Class<URI> getMetadataType() {
-            return URI.class;
+        public Set<Class<?>> getMetadataTypes() {
+            return metadataTypes;
         }
 
         @Override
         public SimpleHtmlWriter render(Object metadata, SimpleHtmlWriter htmlWriter) throws IOException {
-            return (SimpleHtmlWriter) htmlWriter.startElement("a").attribute("href", metadata.toString())
-                .characters(MetadataRenderer.trimIfNecessary(metadata.toString()))
+            String text;
+            String link;
+            if (metadata instanceof File) {
```

Comment:
It could be nice to refactor this into a File handling method that both this branch and RegularFile can use, but not necessary.

Reply:
Yeah, I wish this was more straightforward.  I'm always surprised at the lack of relationship between `File` and `RegularFile`.